cmd/kubeadm/app/preflight/checks.go

```diff
@@ -351,6 +353,24 @@ func (sysver SystemVerificationCheck) Check() (warnings, errors []error) {
 	return warns, nil
 }
 
+type KubernetesVersionCheck struct {
+	KubernetesVersion string
+}
+
+func (kubever KubernetesVersionCheck) Check() (warnings, errors []error) {
+	k8sVersion, err := version.ParseSemantic(kubever.KubernetesVersion)
+	if err != nil {
+		return nil, []error{fmt.Errorf("couldn't parse kubernetes version %q: %v", kubever.KubernetesVersion, err)}
+	}
+
+	kubeadmVersion := apiversion.Get()
+	if compare, _ := k8sVersion.Compare(kubeadmVersion.GitVersion); compare == 1 {
```

Comment:
Could you only compare minor versions? That is `v1.x.4` vs `v1.y.6` => `if x > y { // do things } `

I don't want to see a warning everytime I start up Kubernetes `v1.6.5` on kubeadm `v1.6.4`...
We actually indirectly support the Kubernetes `v1.6.5` on kubeadm `v1.6.4` case indirectly from the [Kubernetes Versioning guidelines](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/versioning.md).